account/src/main/java/com/caldevsupplychain/account/controller/AccountControllerImpl.java

```diff
@@ -84,24 +77,149 @@
 			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ACCOUNT_EXIST.name(), "Account already registered."), HttpStatus.CONFLICT);
 		}
 
-		UserBean userBean = userMapper.userWSToBean(userWS);
+		UserBean userBean = userMapper.toBean(userWS);
 
 		UserBean user = accountService.createUser(userBean);
 
 		try {
-			emailService.sendVerificationTokenEmail(user.getEmailAddress(), user.getToken(), EmailType.REGISTRATION.name());
+			emailService.sendVerificationTokenEmail(user.getEmailAddress(), user.getToken(), EmailType.ACTIVATION.name());
+		} catch (MessagingException e) {
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.EMAIL_MESSAGING_EXCEPTION.name(), e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		return new ResponseEntity<>(userMapper.toWS(user), HttpStatus.CREATED);
+	}
+
+	@GetMapping("/activate/{token}")
+	public ResponseEntity<?> activateAccount(@PathVariable("token") String token) {
+		Optional<UserBean> user = accountService.findByToken(token);
+
+		if (!user.isPresent()) {
+			return new ResponseEntity<Object>(new ApiErrorsWS(ErrorCode.INVALID_TOKEN.name(), "Invalid Token."), HttpStatus.BAD_REQUEST);
+		}
+		accountService.activateUser(user.get().getId());
+
+		return new ResponseEntity<>(userMapper.toWS(user.get()), HttpStatus.OK);
+	}
+
+	@PostMapping("/issue-token")
+	@RequiresAuthentication
+	public ResponseEntity<?> issueToken(@Validated @RequestBody UserWS userWS) {
+
+		Optional<UserBean> user = accountService.findByEmailAddress(userWS.getEmailAddress());
+
+		if (!user.isPresent()) {
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.LOGIN_INVALID.name(), "Invalid Login"), HttpStatus.UNAUTHORIZED);
+		}
+
+		Subject subject = SecurityUtils.getSubject();
+
+		// Apache Shiro authentication check for 1st login
+		if (!subject.isAuthenticated()) {
+			UsernamePasswordToken token = new UsernamePasswordToken(userWS.getEmailAddress(), userWS.getPassword());
+
+			try {
+				subject.login(token);
+			} catch (AuthenticationException e) {
+				log.error("Error in login. User fail to login. user={}", user.toString());
+				return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.LOGIN_INVALID.name(), e.getMessage()), HttpStatus.UNAUTHORIZED);
+			}
+		}
+
+		log.info("Success in user login. user={}", user.toString());
+
+		UserBean userBean = user.get();
+
+		JWTAuthenticationToken jwtObj = jwtService.createJwtToken(userBean);
+
+		log.info("Success create jwt token. jwtToken={}", jwtObj.getToken());
+
+		return new ResponseEntity<>(jwtService.createJwtHeader(jwtObj.getToken()), HttpStatus.OK);
+	}
+
+
+	@GetMapping("/revoke-token")
+	@RequiresAuthentication
+	public ResponseEntity<?> revokeToken() {
```

Comment:
Looks like there's no need to revoke a token (https://github.com/auth0/node-jsonwebtoken/issues/103). FE logout should be just simply remove token from the session. So you can remove this end point.

Reply:
fixed